src/plot_api/subroutines.js

```diff
@@ -113,11 +110,6 @@ function lsInner(gd) {
         ax._mainMirrorPosition = (ax.mirror && counterAx) ?
             getLinePosition(ax, counterAx,
                 alignmentConstants.OPPOSITE_SIDE[ax.side]) : null;
-
-        // Figure out which subplot to draw ticks, labels, & axis lines on
-        // do this as a separate loop so we already have all the
-        // _mainAxis and _anchorAxis links set
-        ax._mainSubplot = findMainSubplot(ax, fullLayout);
```

Comment:
moved this into `plots.linkSubplots`, which is both a more obvious place for it and happens earlier (in `supplyDefaults`) so `_mainSubplot` is available sooner.